File: service/app/src/main/java/io/miragon/blueprint/adapter/inbound/operaton/IssueInsurancePolicyWorker.java
package io.miragon.blueprint.adapter.inbound.operaton;

import dev.bpmcrafters.processengine.worker.ProcessEngineWorker;
import dev.bpmcrafters.processengine.worker.Variable;
import io.miragon.blueprint.adapter.process.BikeLeasingProcessProcessApi.ServiceTasks;
import io.miragon.blueprint.application.port.inbound.IssueInsurancePolicyUseCase;
import io.miragon.blueprint.domain.leasing.ApplicationId;
import org.springframework.stereotype.Component;

@Component
public class IssueInsurancePolicyWorker {

    private final IssueInsurancePolicyUseCase useCase;

    public IssueInsurancePolicyWorker(IssueInsurancePolicyUseCase useCase) {
        this.useCase = useCase;
    }

    @ProcessEngineWorker(topic = ServiceTasks.ISSUE_INSURANCE_POLICY)
    public void issueInsurancePolicy(@Variable String applicationId) {
        useCase.issuePolicy(ApplicationId.of(applicationId));
    }
}

File: service/app/src/main/kotlin/io/miragon/blueprint/adapter/inbound/operaton/IssueInsurancePolicyWorker.kt
package io.miragon.blueprint.adapter.inbound.operaton

import dev.bpmcrafters.processengine.worker.ProcessEngineWorker
import dev.bpmcrafters.processengine.worker.Variable
import io.miragon.blueprint.adapter.process.BikeLeasingProcessProcessApi.ServiceTasks
import io.miragon.blueprint.application.port.inbound.IssueInsurancePolicyUseCase
import io.miragon.blueprint.domain.leasing.ApplicationId
import org.springframework.stereotype.Component

@Component
class IssueInsurancePolicyWorker(
    private val useCase: IssueInsurancePolicyUseCase,
) {

    @ProcessEngineWorker(topic = ServiceTasks.ISSUE_INSURANCE_POLICY)
    fun issueInsurancePolicy(@Variable applicationId: String) {
        useCase.issuePolicy(ApplicationId.of(applicationId))
    }
}
